modules/encoders/php/hex.rb

```diff
@@ -0,0 +1,57 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Encoder
+  Rank = GreatRanking
+
+  def initialize
+    super(
+      'Name' => 'PHP Hex Encoder',
+      'Description' => %q{
+        This encoder returns a hex string encapsulated in
+        eval(hex2bin()), increasing the size by a bit more than
+        a factor two.
+      },
+      'Author' => 'Julien Voisin',
+      'License' => BSD_LICENSE,
+      'Arch' => ARCH_PHP)
+  end
+
+  def bin_to_hex(str)
+    str.each_byte.map { |b| b.to_s(16) }.join
```

Comment:
This has bitten me in the past. It yields incorrect results when the base-16 representation only needs a single digit. I recommend switching to `.unpack1('H*')`.

```
[1] pry(main)> "\x01\x02\x03".b.each_byte.map { |b| b.to_s(16) }.join
=> "123"
[2] pry(main)> "\x01\x02\x03".b.unpack1('H*')
=> "010203"
[3] pry(main)> 
```